lldb/source/Plugins/Language/Swift/SwiftFormatters.cpp

```diff
@@ -95,10 +116,11 @@ static bool readStringFromAddress(
       StringPrinter::StringElementType::UTF8>(read_options);
 };
 
-bool lldb_private::formatters::swift::StringGuts_SummaryProvider(
+static bool makeStringGutsSummary(
     ValueObject &valobj, Stream &stream,
     const TypeSummaryOptions &summary_options,
-    StringPrinter::ReadStringAndDumpToStreamOptions read_options) {
+    StringPrinter::ReadStringAndDumpToStreamOptions read_options,
+    StringSlice slice = {0, UINT64_MAX}) {
```

Comment:
If I'm reading this correctly, this default slice of `{0, UINT64_MAX}` won't do anything in `applySlice`, so do you think this could be an optional with a default case of `None`, and before we call `applySlice` we check if there's a slice to apply or not?

Reply:
I started with `Optional`, and switched to this, because subjectively I felt `Optional` didn't add much. The bounds still need to be checked. In the `Optional` case, both `None` and bounds need to be checked, in the way it currently is, only the bounds needs to be checked. Using a default of "max slice"  I think results in less logic. I can go back that way, if we think it's better to be more explicit.

Reply:
I personally prefer `Optional` because to figure out what happens in the default max slice case you have to run the `applySlice` function in your head to figure out that it doesn't do anything. This is a nit though and I'm fine with either.

Reply:
let's meet in the middle: `llvm::Optional<StringSlice> slice = {{0, UINT64_MAX}}` :trollface:

Reply:
Now using `Optional`